libs/dyn/merge/override.go

```diff
@@ -13,11 +13,17 @@ import (
 // For instance, it can disallow changes outside the specific path(s), or update
 // the location of the effective value.
 //
+// Values returned by 'VisitDelete', 'VisitInsert' and 'VisitUpdate' are used as
+// the final value of the node. 'VisitDelete' should return dyn.NilValue to
+// do the delete, or can return 'left' to undo it.
+//
+// TODO 'VisitDelete' and 'VisitInsert' should support dyn.NilValue as well
```

Comment:
What would the semantics be?

I think you mean `VisitUpdate` here, not `VisitDelete`.